eng/native/configureplatform.cmake

```diff
@@ -214,6 +214,10 @@ if(CLR_CMAKE_HOST_OS STREQUAL Emscripten)
     set(CLR_CMAKE_HOST_BROWSER 1)
 endif(CLR_CMAKE_HOST_OS STREQUAL Emscripten)
 
+if(CLR_CMAKE_TARGET_OS STREQUAL Wasi)
+    set(CLR_CMAKE_HOST_WASI 1)
```

Comment:
I think you'll need `set(CLR_CMAKE_HOST_UNIX 1)` here too

Reply:
Nah, WASI is not really unix

Reply:
I know, the same applies for Emscripten but I think some other parts of the build required it. Maybe that's no longer the case though?

Reply:
This is just initial take on the whole build to get start things moving. 
I have another PR on which I have more progress.
Let's return to it on one of the next PRs